packages/slate-react/src/utils/dom.ts

```diff
@@ -314,3 +314,18 @@ export const isTrackedMutation = (
   // Target add/remove is tracked. Track the mutation if we track the parent mutation.
   return isTrackedMutation(editor, parentMutation, batch)
 }
+
+export function getActiveElement() {
+  let active = document.activeElement
+
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
```

Comment:
why while(true)?

Reply:
It is for allowing deep traversal through nested shadow roots to find the actual active element. I think [b1710e0](https://github.com/ianstormtaylor/slate/pull/5648/commits/b1710e0adc10dc0054639a66db39d6f22d1f734d) version is better as it places the continuation conditions in the loop to make it visible and understandable.